pkg/apis/operator/v1alpha1/glustercluster_types.go

```diff
@@ -1,22 +1,60 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// GlusterStorageTarget defines a storage target
+type GlusterStorageTarget struct {
+	Name    string       `json:"name,omitempty"`
+	Address []string     `json:"address"`
+	Creds   *Credentials `json:"credentials,omitempty"`
+}
+
+// GlusterClusterReplicationDetails defines replication details
+type GlusterClusterReplicationDetails struct {
+	Creds   *Credentials           `json:"credentials,omitempty"`
+	Targets []GlusterStorageTarget `json:"targets"`
+}
+
+// GlusterNodeThreshold defines threshold details for node
+type GlusterNodeThreshold struct {
+	Nodes          *int   `json:"nodes,omitempty"`
+	MinNodes       *int   `json:"minNodes,omitempty"`
+	MaxNodes       *int   `json:"maxNodes,omitempty"`
+	FreeStorageMin string `json:"freeStorageMin,omitempty"`
+	FreeStorageMax string `json:"freeStorageMax,omitempty"`
+}
+
+// GlusterNodeStorageDetails defines storage class details
+type GlusterNodeStorageDetails struct {
+	StorageClassName string             `json:"storageClassName,omitempty"`
+	Capacity         *resource.Quantity `json:"capacity,omitempty"`
+}
+
+// GlusterNodeTemplate defines a gluster node's template
+type GlusterNodeTemplate struct {
+	Name       string                     `json:"name,omitempty"`
+	Zone       string                     `json:"zone,omitempty"`
+	Thresholds *GlusterNodeThreshold      `json:"thresholds,omitempty"`
```

Comment:
May be Threshold

Reply:
Under thresholds we have different values like `Nodes`, `MinNodes`, `MaxNodes`, `FreeStorageMin` and `FreeStorageMax` so kept it like this.

Reply:
Let's go ahead an make this singular.

Reply:
Ack